.circleci/config.yml

```diff
@@ -158,13 +160,16 @@ workflows:
             branches:
               ignore: /.*/
             tags:
-              only: /^v?[0-9]+(\.[0-9]+)*(?:-stage)?$/
+              # match tags such as "2022.03.08" or "2022.03.08-1", with or
+              # without a -stage suffix.
+              only: /^20\d{2}\.\d{2}\.\d{2}(?:-\d+)?(?:-stage)?$/
       - deploy:
           requires:
             - build
           filters:
             branches:
               ignore: /.*/
             tags:
-              # match tags such as "v1.1.1" or "v1.1.1-stage"
-              only: /^v?[0-9]+(\.[0-9]+)*(?:-stage)?$/
+              # match tags such as "2022.03.08" or "2022.03.08-1", with or
+              # without a -stage suffix.
+              only: /^20\d{2}\.\d{2}\.\d{2}(?:-\d+)?(?:-stage)?$/
```

Comment:
I don't suppose there is a way in yaml to define this regex once in the file so we don't have to repeat it?

Reply:
Not trivially as far as I'm aware